state/api/src/main/java/com/expediagroup/streamplatform/streamregistry/state/EntityView.java

```diff
@@ -63,4 +63,25 @@
    * @return a stream containing all entities of the given key type.
    */
   <K extends Key<S>, S extends Specification> Stream<Entity<K, S>> all(Class<K> keyClass);
+
+  /**
+   * Returns a {@link Stream} containing all entities of the given {@link Key} type which have been deleted but not
+   * purged ({@link #purgeDeleted(Key)})
+   *
+   * @param keyClass the key class of an entity type.
+   * @param <K>      the key type.
+   * @param <S>      the specification type.
+   * @return a stream containing all deleted entities of the given key type.
+   */
+  <K extends Key<S>, S extends Specification> Stream<Entity<K, S>> allDeleted(Class<K> keyClass);
+
+  /**
+   * Purge a deleted entity so that it no longer appears in {@link #allDeleted(Class)}.
+   *
+   * @param key the key of the entity to be purged
+   * @param <K> the key type.
+   * @param <S> the specification type.
+   * @return an optional containing the entity that was purged or empty if no entity exists.
+   */
+  <K extends Key<S>, S extends Specification> Optional<Entity<K, S>> purgeDeleted(K key);
```

Comment:
Stream Registry itself uses this API in the repository module and it currently has no need to react to events. Now it does - to purge deleted items from the view.

The `entityView` bean in `KafkaConfiguration` in the `repository-kafka` module will need an `EntityViewListener` implementation that calls `purgeDeleted` for `SpecificationDeletionEvent`s. 

The `EntityView#load()` can also probably be removed.

Reply:
Great, that makes sense. I wasn't too sure how was best to handle the purge in Stream Registry.

Reply:
resolved with `PurgingEntityViewListener`